robot-core/src/main/java/org/obolibrary/robot/template/Template.java

```
@@ -0,0 +1,1852 @@
+
+    String propertyType = null;
+    if (propertyTypeColumn != 1) {
+      try {
+        propertyType = row.get(propertyTypeColumn);
+      } catch (IndexOutOfBoundsException e) {
+        // do nothing
+      }
+    }
+    List<String> propertyTypes =
+        TemplateHelper.getTypes(propertyType, propertyTypeSplit, "subproperty");
+
+    OWLObjectProperty property = dataFactory.getOWLObjectProperty(iri);
+
+    // Handle special property types
+    for (String pt : propertyTypes) {
+      switch (pt.trim().toLowerCase()) {
+        case "asymmetric":
+          axioms.add(dataFactory.getOWLAsymmetricObjectPropertyAxiom(property));
+          break;
+        case "functional":
+          axioms.add(dataFactory.getOWLFunctionalObjectPropertyAxiom(property));
+          break;
+        case "inversefunctional":
+        case "inverse functional":
+          axioms.add(dataFactory.getOWLInverseFunctionalObjectPropertyAxiom(property));
+          break;
+        case "irreflexive":
+          axioms.add(dataFactory.getOWLIrreflexiveObjectPropertyAxiom(property));
+          break;
+        case "reflexive":
+          axioms.add(dataFactory.getOWLReflexiveObjectPropertyAxiom(property));
+          break;
+        case "symmetric":
+          axioms.add(dataFactory.getOWLSymmetricObjectPropertyAxiom(property));
+          break;
+        case "transitive":
+          axioms.add(dataFactory.getOWLTransitiveObjectPropertyAxiom(property));
+          break;
+      }
+    }
+    // Remove the special types
+    propertyTypes.removeAll(
+        Lists.newArrayList(
+            "asymmetric",
+            "functional",
+            "inversefunctional",
+            "inverse functional",
+            "irreflexive",
+            "reflexive",
+            "symmetric",
+            "transitive"));
+    if (propertyTypes.size() > 1) {
+      // TODO - can only have one of: subproperty, equivalent, disjoint, or inverse
+      throw new Exception();
```

Please throw a nicer exception, with a description and link to the docs.